### ISSUE TITLE
*Seeking Contribution* [FIX] Generate previews from forked repos. This currently doesn't work.

### ISSUE THREAD

**Issue by stoobie** [starknet-io/starknet-docs#1043]
# Background

When you create or update a PR, a preview of the site is generated with your changes and deployed to the `gh-pages` branch in a directory named after the PR ID.

# Problem

The Github flow `deploy-review` in `/.github/workflows/pull_request.yml` does not successfully deploy the generated site preview to the `gh-pages` branch for contributions originating from forks. The issue becomes visible when the runner runs the[ _Commit changes_](https://github.com/starknet-io/starknet-docs/blob/63c5c84b176e0dfa47bbb28e1ff1484e6f667835/.github/workflows/pull_request.yml#L75) step in the pull_requent.yml workflow.

# To Reproduce

1. Fork (and optionally clone) the starknet-docs repo.
2. Create a new branch and make a small change in an .adoc file.
3. Push the new branch and create a new PR.

# Expected behavior
You should see this in **Conversation** tab of the PR
![image](https://github.com/starknet-io/starknet-docs/assets/39279277/b3057536-3332-487b-8120-e75a988f314c)

You should also see a link like this one (with the correct PR-number):
![image](https://github.com/starknet-io/starknet-docs/assets/39279277/6483335a-6d7c-4824-a8db-4607c9046f55)

**Comment by xiaolou86** [comment]
@stoobie Can I take the chance to try to fix it? And it will have the `Remuneration provided` label :(

**Comment by xiaolou86** [comment]
But I don't have related experiences, I need to learn more while doing :(

**Comment by stoobie** [comment]
@xiaolou86 I did some research yesterday, and if I understand correctly, the problem lies in the token that the `deploy` workflow uses to commit the preview site to the gh-pages branch on starknet-io/starknet-docs.

I think that the only workaround that still uses Github pages is to generate the preview on the gh-pages branch of the forked repo (user/starknet-docs), not on starknet-io/starknet-docs.

Regarding remuneration for this task, I need to confer with a couple of stakeholders.

**Comment by xiaolou86** [comment]
> @xiaolou86 I did some research yesterday, and if I understand correctly, the problem lies in the token that the `deploy` workflow uses to commit the preview site to the gh-pages branch on starknet-io/starknet-docs.
> 
> I think that the only workaround that still uses Github pages is to generate the preview on the gh-pages branch of the forked repo (user/starknet-docs), not on starknet-io/starknet-docs.
> 
> Regarding remuneration for this task, I need to confer with a couple of stakeholders.

I agree with you. It looks like something matter with the token, because it works well only for Collaborator.

**Comment by stoobie** [comment]
> I agree with you. It looks like something matter with the token, because it works well only for Collaborator.

I do think it's worth looking into the workaround I mentioned: Publishing the preview on the gh-pages branch of the forked repo (user/starknet-docs), not on starknet-io/starknet-docs.

**Comment by xiaolou86** [comment]
@stoobie in the preview's workflow, what's the purpose to commit changes to `gh-pages` branch?

**Comment by xiaolou86** [comment]
I didn't see the step to deploy web site in the workflow

**Comment by xiaolou86** [comment]
In the github's introduction, there are 3 steps:
1. upload-artifact
2. download-artifact
3. deploy

**Comment by stoobie** [comment]
The Preview site is the gh-pages branch. Pushing to that branch is what
publishes (deploys) the preview.



*Steve Goodman*

*Principal Technical Writer*
*+972-544301982*
*https://starkware.co <https://starkware.co/>*

Sent via Superhuman ***@***.***>

**Comment by xiaolou86** [comment]
@stoobie I looked into many docs and tried many times... at last, I think I find one solution. 

Details in the following.
1. Change the event from `pull_request` to `pull_request_target`. [permissions-for-the-github_token](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token)

2. Change the branch ref when checkout. Because its event (pull_request_target) runs in the context of the base of the pull request, rather than in the context of the merge commit as the pull_request event does. [base branch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target)

3. Since of the above 2, the workflow triggered from `pull_request_target` event will run only after the .yml config has been merged into repo.

**Comment by xiaolou86** [comment]
The first time, workflow triggered from pull_request_target event won't run until this .yml config has been merged into repo.
After that, the workflow will run when event triggered.

**Comment by stoobie** [comment]
@xiaolou86 Thanks for your hard work on this! Reward sent through OnlyDust.xyz. Please make sure you received it.

**Comment by xiaolou86** [comment]
@stoobie I saw the reward being processed. Thank you very much for recognizing my work.
